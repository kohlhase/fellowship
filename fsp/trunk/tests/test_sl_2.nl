Parameters A,B: bool
we need to prove ¬(A-B)⇒¬B⇒¬A
  assume ¬(A-B) (nab)
  assume ¬B (nb)
  we need to prove ¬A
    assume A (H2)
    we need to prove ⊥
      by H2
      we proved A (a)
      by nab
      we proved ¬(A-B) (H1)
      by H1
      and <???
        we proved B (b)
        by nb
        we proved ¬B (H1)
        by H1
        and by b
      absurd
        by a
      ???>
    absurd
  done
done
