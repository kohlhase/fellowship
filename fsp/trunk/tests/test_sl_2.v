(* Coq script *) 
Parameter A B : Prop. 
Theorem t1 : (~(A /\ ~B) -> (~B -> ~A)). 
Proof (fun (nab:~(A /\ ~B)) => (fun (nb:~B) => (fun (H2:A) => (let a:A := H2 in (let H1:~(A /\ ~B) := nab in (match (H1 (conj a (fun x1 => (let b:B := x1 in (let H1:~B := nb in (match (H1 b) with end)))))) with end)))))). 
