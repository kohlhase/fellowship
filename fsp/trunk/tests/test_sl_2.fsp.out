
    > Current logic: intuitionistic sequent calculus.
      

    > Current logic: intuitionistic sequent calculus.
      

    > Current logic: intuitionistic sequent calculus.
      
fsp < 
    > A,B defined.
      
fsp < 
    > Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<?1||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        .... (1) <======
      done
      
      1 goal yet to prove!
      
      
       |-----  1
      *:¬(A-B)⇒¬B⇒¬A
      
fsp < 
    > Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).?1.1||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        .... (1.1) <======
      done
      
      1 goal yet to prove!
      
      
      nab:¬(A-B)
       |-----  1.1
      *:¬B⇒¬A
      
fsp < 
    > Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.?1.1.1||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        .... (1.1.1) <======
      done
      
      1 goal yet to prove!
      
      
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1
      *:¬A
      
fsp < 
    > Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||?1.1.1.1>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
          ...(1.1.1.1) <======
        done
      done
      
      1 goal yet to prove!
      
      
      *:A
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1.1
      
fsp < 
    > 2 subgoals generated. 
      
      Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<?1.1.1.1.1||?1.1.1.1.2>>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            .... (1.1.1.1.1) <======
          ...(1.1.1.1.2)
        done
      done
      
      2 goals yet to prove!
      
      
      a:A
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1.1.1
      *:¬(A-B)
      
fsp < 
    > Closed a branch. 
      
      Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<nab||?1.1.1.1.2>>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            by nab
          ...(1.1.1.1.2) <======
        done
      done
      
      1 goal yet to prove!
      
      
      *:¬(A-B)
      a:A
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1.1.2
      
fsp < 
    > Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<nab||μ'H1:¬(A-B).<H1||?1.1.1.1.2.1*_F_>>>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            by nab
            we proved ¬(A-B) (H1)
            by H1
            and .... (1.1.1.1.2.1) <======
          absurd
        done
      done
      
      1 goal yet to prove!
      
      
      a:A
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1.1.2.1
      *:A-B
      
fsp < 
    > 2 subgoals generated. 
      
      Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<nab||μ'H1:¬(A-B).<H1||?1.1.1.1.2.1.1*?1.1.1.1.2.1.2*_F_>>>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            by nab
            we proved ¬(A-B) (H1)
            by H1
            and <???
            ...(1.1.1.1.2.1.1) <======
              .... (1.1.1.1.2.1.2)
            ???>
          absurd
        done
      done
      
      2 goals yet to prove!
      
      
      *:B
      a:A
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1.1.2.1.1
      
fsp < 
    > 2 subgoals generated. 
      
      Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<nab||μ'H1:¬(A-B).<H1||μ'b:B.<?1.1.1.1.2.1.1.1||?1.1.1.1.2.1.1.2>*?1.1.1.1.2.1.2*_F_>>>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            by nab
            we proved ¬(A-B) (H1)
            by H1
            and <???
              we proved B (b)
              .... (1.1.1.1.2.1.1.1) <======
            ...(1.1.1.1.2.1.1.2)
              .... (1.1.1.1.2.1.2)
            ???>
          absurd
        done
      done
      
      3 goals yet to prove!
      
      
      b:B
      a:A
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1.1.2.1.1.1
      *:¬B
      
fsp < 
    > Closed a branch. 
      
      Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<nab||μ'H1:¬(A-B).<H1||μ'b:B.<nb||?1.1.1.1.2.1.1.2>*?1.1.1.1.2.1.2*_F_>>>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            by nab
            we proved ¬(A-B) (H1)
            by H1
            and <???
              we proved B (b)
              by nb
            ...(1.1.1.1.2.1.1.2) <======
              .... (1.1.1.1.2.1.2)
            ???>
          absurd
        done
      done
      
      2 goals yet to prove!
      
      
      *:¬B
      b:B
      a:A
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1.1.2.1.1.2
      
fsp < 
    > Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<nab||μ'H1:¬(A-B).<H1||μ'b:B.<nb||μ'H1:¬B.<H1||?1.1.1.1.2.1.1.2.1*_F_>>*?1.1.1.1.2.1.2*_F_>>>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            by nab
            we proved ¬(A-B) (H1)
            by H1
            and <???
              we proved B (b)
              by nb
              we proved ¬B (H1)
              by H1
              and .... (1.1.1.1.2.1.1.2.1) <======
            absurd
              .... (1.1.1.1.2.1.2)
            ???>
          absurd
        done
      done
      
      2 goals yet to prove!
      
      
      b:B
      a:A
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1.1.2.1.1.2.1
      *:B
      
fsp < 
    > Closed a branch. 
      
      Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<nab||μ'H1:¬(A-B).<H1||μ'b:B.<nb||μ'H1:¬B.<H1||b*_F_>>*?1.1.1.1.2.1.2*_F_>>>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            by nab
            we proved ¬(A-B) (H1)
            by H1
            and <???
              we proved B (b)
              by nb
              we proved ¬B (H1)
              by H1
              and by b
            absurd
              .... (1.1.1.1.2.1.2) <======
            ???>
          absurd
        done
      done
      
      1 goal yet to prove!
      
      
      a:A
      nb:¬B
      nab:¬(A-B)
       |-----  1.1.1.1.2.1.2
      *:A
      
fsp < 
    > Closed the last branch: 
      Proof completed!
      
      Proof term: 
      μthesis:¬(A-B)⇒¬B⇒¬A.<λnab:¬(A-B).λnb:¬B.μH1:¬A.<λH2:A.μH3:⊥.<H2||μ'a:A.<nab||μ'H1:¬(A-B).<H1||μ'b:B.<nb||μ'H1:¬B.<H1||b*_F_>>*a*_F_>>>||H1>||thesis>
      Natural language:
      we need to prove ¬(A-B)⇒¬B⇒¬A
        assume ¬(A-B) (nab)
        assume ¬B (nb)
        we need to prove ¬A
          assume A (H2)
          we need to prove ⊥
            by H2
            we proved A (a)
            by nab
            we proved ¬(A-B) (H1)
            by H1
            and <???
              we proved B (b)
              by nb
              we proved ¬B (H1)
              by H1
              and by b
            absurd
              by a
            ???>
          absurd
        done
      done
      
      
fsp < 
    > t1 defined.
      
fsp < fsp < 
    > Firing up Coq for confirmation.
      
fsp < 
    > Firing up Coq for confirmation.
      
fsp < 
    > File ./test_sl_2.nl written. Enjoy the reading.
      
fsp < 
    > EOF!

